docs/DeveloperGuide.md

```diff
@@ -270,21 +270,58 @@ The constraints above have been applied after careful consideration of the needs
 * The `value` attribute in the Quantity field has been stored as a `double` value.
 * Users will have different kinds of units while tracking the quantity of their food items. For example, kg, can, bottle etc.
 * To ensure the user's freedom in choosing the unit, the `unit` attribute in the Quantity class has been stored as a `String` field.
-* Constraints on the content of the `unit` field have been applied to prevent misuse of the feature. 
+* Constraints on the content of the `unit` field have been applied to prevent misuse of the feature.
 * Units are always single words and do not contain special characters or numbers. Hence, these restrictions have been enforced while accepting a quantity from the user.
-* The `unit` field is optional to relieve the user of extra work of giving a unit in intuitive cases. For example, while storing the quantity of fish a unit is intuitively the number of fish. 
+* The `unit` field is optional to relieve the user of extra work of giving a unit in intuitive cases. For example, while storing the quantity of fish a unit is intuitively the number of fish.
 
 #### Design Consideration:
 
 ##### Aspect: Implementation
 
 * <b>Alternative 1 (current choice): </b> Value and unit stored as double and string fields in the Quantity class.
-    * Pros: Easy to implement. 
+    * Pros: Easy to implement.
     * Cons: Less OOP compared to the other alternative.
 * <b>Alternative 2: </b> Value and Unit extracted to different classes. Quantity class' attributes are objects of these Value and Unit classes.
     * Pros: Improves OOP aspect of the code.
     * Cons: Unnecessarily complicates the code.
 
+### Find feature
+
+The find feature allows users to search for food items based on description, expiration date, priority and/or tags.
+
+#### Implementation:
+
+The find feature is achieved by setting a `predicate` on the `filteredList` located in `ModelManager`. This can be done using the method `ModelManager#updateFilteredFoodList(Predicate<Food> predicate)`, which uses the parameter as the `predicate`. There are 4 valid `predicate`, namely `DescriptionContainsKeywordsPredicate`, `ExpirySearchPredicate`, `PrioritySearchPredicate` and `TagSearchPredicate`, all of which implements the `Predicate<Food>` interface. Based on the user's `find` command, each of the `predicate` is generated with the relevant search parameters. Then, the `FindCommand#combinePredicates()` method will combine all the predicates into a single `predicate` and is passed to `ModelManager#updateFilteredFoodList(Predicate<Food> predicate)`.
```

Comment:
I think it should be "all of which implement"

Reply:
Noted. Will change.